src/types/redis_string.h

```diff
@@ -42,8 +44,34 @@ struct StringSetArgs {
   bool keep_ttl;
 };
 
-namespace redis {
+enum class StringLCSType { NONE, LEN, IDX };
+
+struct StringLCSArgs {
+  StringLCSType type;
+  int64_t min_match_len;
+};
 
+struct StringLCSRange {
+  uint32_t start;
+  uint32_t end;
+};
+
+struct StringLCSMatchedRange {
+  StringLCSRange a;
+  StringLCSRange b;
+  uint32_t match_len;
+};
+
+struct StringLCSIdxResult {
+  // Matched ranges.
+  std::list<StringLCSMatchedRange> matches;
```

Comment:
Though I don't think performance of LCS it's important, I think `deque` or `vector` ( using vector without pop_front) would be better here.

Reply:
Yeah, deque or vector is better here.

Reply:
Thank. I have modified this to use the `vector`.

Initially, I hadn't used the `vector` because I was concerned that a very large `vector` would significantly increase memory usage when building the `output` at the end.
However, I forgot that `dp` actually frees up more memory space at the end of `string_db.LCS()`.